### PR TITLE
Add Ploopyco functions for host state control

### PR DESCRIPTION
**Pull request by t4corun**
Addressed gap in Ploopy Nano where drag scroll and dpi cannot be accessed at the userspace level. User can now implement Host State Control in their keymap

## Description

Extends on the current code base. Drag scroll variable and dpi cycling logic moved into functions so they can be called at the userspace level.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

N/A 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a chano the documentation.ge to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by t4corun** [comment]
Revised PR based on @obosob's Discord comment. Removed keyboard level Host State control. Leave it to the user to implement it at the keymap. Tested on my Ploopy Nano.

The previous commit still has the Host state control if that is still desired.

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
> Revised PR based on obosob's Discord comment. Removed keyboard level Host State control. Leave it to the user to implement it at the keymap. Tested on my Ploopy Nano.

Agree with obosob. This is something best left implemented at the user level. 

However the helper functions are a good idea and useful.

**Comment by t4corun** [comment]
Thanks, but credit to @obosob for the idea. I was originally trying to call the ploopyco DRAG_SCROLL from my other QMK boards and was struggling

**Comment by t4corun** [comment]
@drashna Maybe I made a mistake when I created this PR, was I supposed to submit this to qmk:develop? I think this wasn't included in the 0.26 release. Thanks

**Comment by drashna** [comment]
It should probably go to develop, yeah.  But given the parts that it is touching... it may be fine hitting master directly.

**Comment by t4corun** [comment]
Updated the PR to hit develop instead. Will keep this in mind for next time. Thanks again

File: keyboards/ploopyco/ploopyco.c [hunk 1]
```diff
@@ -68,6 +68,7 @@ bool  is_drag_scroll       = false;
 float scroll_accumulated_h = 0;
 float scroll_accumulated_v = 0;
 
+
```
Comment (fauxpark) [inline keyboards/ploopyco/ploopyco.c:71]:
```suggestion
```

File: keyboards/ploopyco/ploopyco.c [hunk 2]
```diff
@@ -128,6 +129,18 @@ void encoder_driver_task(void) {
 }
 #endif
 
+
```
Comment (fauxpark) [inline keyboards/ploopyco/ploopyco.c:132]:
```suggestion
```

File: keyboards/ploopyco/ploopyco.c [hunk 3]
```diff
@@ -128,6 +129,18 @@ void encoder_driver_task(void) {
...
+    is_drag_scroll ^= 1;
+}
+
+void cycle_dpi(void) {
+    keyboard_config.dpi_config = (keyboard_config.dpi_config + 1) % DPI_OPTION_SIZE;
+    eeconfig_update_kb(keyboard_config.raw);
+    pointing_device_set_cpi(dpi_array[keyboard_config.dpi_config]);
+}
+
+
```
Comment (fauxpark) [inline keyboards/ploopyco/ploopyco.c:143]:
```suggestion
```

File: keyboards/ploopyco/readme.md [hunk 4]
```diff
@@ -10,11 +10,12 @@
...
-| `PLOOPY_SCROLL_BUTTON_DEBOUNCE` | `100`             | Time to ignore scroll events after pressing scroll wheel. |
+| Define                          | Default Value     | Description                                                                                                                                           |
+| ------------------------------- | ----------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `PLOOPY_IGNORE_SCROLL_CLICK`    | *__not_defined__* | Ignores scroll wheel if it is pressed down.                                                                                                           |
+| `PLOOPY_SCROLL_DEBOUNCE`        | `5`               | Number of milliseconds between scroll events.                                                                                                         |
+| `PLOOPY_SCROLL_BUTTON_DEBOUNCE` | `100`             | Time to ignore scroll events after pressing scroll wheel.                                                                                             |
+
```
Comment (fauxpark) [inline keyboards/ploopyco/readme.md:18]:
```suggestion
```

File: keyboards/ploopyco/ploopyco.h [hunk 5]
```diff
@@ -37,3 +37,5 @@ enum ploopy_keycodes {
 
 bool encoder_update_user(uint8_t index, bool clockwise);
 bool encoder_update_kb(uint8_t index, bool clockwise);
+void toggle_drag_scroll(void);
+void cycle_dpi(void);
```
Comment (fauxpark) [inline keyboards/ploopyco/ploopyco.h:41]:
```suggestion
void cycle_dpi(void);
```
Keep newline at EOF.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ploopyco/ploopyco.c (hunk 1 above)
Comment (t4corun) [reply keyboards/ploopyco/ploopyco.c]: done!

File: keyboards/ploopyco/ploopyco.c (hunk 2 above)
Comment (t4corun) [reply keyboards/ploopyco/ploopyco.c]: done!

File: keyboards/ploopyco/ploopyco.c (hunk 3 above)
Comment (t4corun) [reply keyboards/ploopyco/ploopyco.c]: done

File: keyboards/ploopyco/readme.md (hunk 4 above)
Comment (t4corun) [reply keyboards/ploopyco/readme.md:18]: completed

File: keyboards/ploopyco/ploopyco.h (hunk 5 above)
Comment (t4corun) [reply keyboards/ploopyco/ploopyco.h]: Stared at this way too long before I saw the "keep newline at EOF". done! Thank you for reviewing

File: keyboards/ploopyco/readme.md [hunk 6]
```diff
@@ -10,11 +10,11 @@
...
-| `PLOOPY_SCROLL_BUTTON_DEBOUNCE` | `100`             | Time to ignore scroll events after pressing scroll wheel. |
+| Define                          | Default Value     | Description                                                                                                                                           |
+| ------------------------------- | ----------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `PLOOPY_IGNORE_SCROLL_CLICK`    | *__not_defined__* | Ignores scroll wheel if it is pressed down.                                                                                                           |
+| `PLOOPY_SCROLL_DEBOUNCE`        | `5`               | Number of milliseconds between scroll events.                                                                                                         |
+| `PLOOPY_SCROLL_BUTTON_DEBOUNCE` | `100`             | Time to ignore scroll events after pressing scroll wheel.                                                                                             |
```
Comment (fauxpark) [inline keyboards/ploopyco/readme.md:17]:
```suggestion
|Define                         |Default Value|Description                                              |
|-------------------------------|-------------|---------------------------------------------------------|
|`PLOOPY_IGNORE_SCROLL_CLICK`   |*Not defined*|Ignores scroll wheel if it is pressed down.              |
|`PLOOPY_SCROLL_DEBOUNCE`       |`5`          |Number of milliseconds between scroll events.            |
|`PLOOPY_SCROLL_BUTTON_DEBOUNCE`|`100`        |Time to ignore scroll events after pressing scroll wheel.|
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ploopyco/readme.md (hunk 6 above)
Comment (t4corun) [reply keyboards/ploopyco/readme.md]: updated

**Review by fauxpark — APPROVED** [review]

**Pull request merged**